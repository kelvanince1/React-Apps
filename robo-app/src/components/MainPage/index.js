import React, { Component } from 'react';

import CardList from '../CardList';
import ErrorBoundary from '../ErrorBoundary';
import Header from '../Header';
import Scroll from '../Scroll';
import SearchBox from '../SearchBox';

class MainPage extends Component {

  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    return this.props.robots.filter(robot => {
        return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
      });
  }

  render() {
    const { isPending, onSearch, robots } = this.props;

    return isPending ?
      <h1>Loading</h1>
    : (
      <div className="tc">
        <Header />
        <SearchBox
          onSearch={onSearch}
        />
        <Scroll>
          <ErrorBoundary>
            <CardList
              robots={this.filterRobots()}
            />
          </ErrorBoundary>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
