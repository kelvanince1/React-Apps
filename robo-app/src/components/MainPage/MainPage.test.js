import React from 'react';
import { shallow } from 'enzyme';

import MainPage from './index';

let wrapper;

beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false 
    }

    wrapper = shallow(<MainPage { ...mockProps } />);
});

it('Expect to render MainPage component', () => {
    expect(wrapper).toMatchSnapshot();
});

it('Filter robots correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [
            {
                id: 3,
                name: 'Johnny',
                email: 'jonhhy@gmail.com'
            }
        ],
        searchField: 'aaa',
        isPending: false 
    }

    const wrapper2 = shallow(<MainPage { ...mockProps2 } />)

    expect(wrapper2.instance().filterRobots([])).toEqual([]);
});
