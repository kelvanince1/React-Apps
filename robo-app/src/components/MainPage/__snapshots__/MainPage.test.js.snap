// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render MainPage component 1`] = `ShallowWrapper {}`;
